From 69b21d8f853cf4b2297ea40b9da0230f6859a99f Mon Sep 17 00:00:00 2001
From: Lin Fuzhen <fuzhen.lin@freescale.com>
Date: Fri, 30 Mar 2012 18:25:17 +0800
Subject: [PATCH 494/678] ENGR00178547 i.mx6dl sabresd: add android config
 file

add android config file;
support booti fastboot command and etc.

Signed-off-by: Lin Fuzhen <fuzhen.lin@freescale.com>
---
 Makefile                                |    1 +
 include/configs/mx6dl_sabresd_android.h |   57 +++++++++++++++++++++++++++++++
 2 files changed, 58 insertions(+), 0 deletions(-)

diff --git a/Makefile b/Makefile
index 411a352..6a4da95 100644
--- a/Makefile
+++ b/Makefile
@@ -3324,6 +3324,7 @@ mx6q_arm2_iram_config	: unconfig
 
 mx6dl_sabresd_config			\
 mx6dl_sabresd_mfg_config		\
+mx6dl_sabresd_android_config			\
 mx6q_sabresd_config			\
 mx6q_sabresd_android_config             \
 mx6q_sabresd_mfg_config			\
diff --git a/include/configs/mx6dl_sabresd_android.h b/include/configs/mx6dl_sabresd_android.h
new file mode 100644
index 0000000..7d836b3
--- /dev/null
+++ b/include/configs/mx6dl_sabresd_android.h
@@ -0,0 +1,57 @@
+/*
+ * Copyright (C) 2010-2012 Freescale Semiconductor, Inc.
+ *
+ * Configuration settings for the MX6DL SabreSD Freescale board.
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU General Public License as
+ * published by the Free Software Foundation; either version 2 of
+ * the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.	 See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 59 Temple Place, Suite 330, Boston,
+ * MA 02111-1307 USA
+ */
+
+#ifndef MX6Q_SABRESD_ANDROID_H
+#define MX6Q_SABRESD_ANDROID_H
+
+#include <asm/arch/mx6.h>
+#include "mx6dl_sabresd.h"
+
+#define CONFIG_CMD_BOOTI
+#define CONFIG_ANDROID_RECOVERY
+/* which mmc bus is your main storage ? */
+#define CONFIG_ANDROID_MAIN_MMC_BUS 3
+
+#define CONFIG_ANDROID_BOOT_PARTITION_MMC 1
+#define CONFIG_ANDROID_SYSTEM_PARTITION_MMC 5
+#define CONFIG_ANDROID_RECOVERY_PARTITION_MMC 2
+#define CONFIG_ANDROID_CACHE_PARTITION_MMC 6
+
+
+#define CONFIG_ANDROID_RECOVERY_BOOTARGS_MMC NULL
+#define CONFIG_ANDROID_RECOVERY_BOOTCMD_MMC  \
+	"booti mmc3 recovery"
+#define CONFIG_ANDROID_RECOVERY_CMD_FILE "/recovery/command"
+
+#undef CONFIG_LOADADDR
+#undef CONFIG_RD_LOADADDR
+#undef CONFIG_EXTRA_ENV_SETTINGS
+
+
+#define CONFIG_LOADADDR		0x10800000	/* loadaddr env var */
+#define CONFIG_RD_LOADADDR      0x11000000
+
+#define CONFIG_INITRD_TAG
+#define	CONFIG_EXTRA_ENV_SETTINGS					\
+		"netdev=eth0\0"						\
+		"ethprime=FEC0\0"					\
+		"bootcmd=booti mmc3\0"
+#endif
-- 
1.7.7.4

